### PR TITLE
Features/auth service

### PR DESCRIPTION
**Pull request by perhanjay**
## 📌 Summary

This PR implements/fixes/adds \[feature/bug] related to \[module/service-name].

## ☑️ Changes Included

This pull request introduces comprehensive improvements to the `auth-service` codebase, focusing on documentation, code quality, and test coverage. The most important changes include the addition of detailed JSDoc documentation guidelines, a new ESLint configuration for code consistency, expanded README documentation, and robust unit tests for key authentication controllers.

**Documentation and Guidelines:**

* Added a `JSDOC_DOCUMENTATION.md` file summarizing completed documentation, providing templates for controllers and validators, listing files pending documentation, and outlining best JSDoc practices for maintainability and IDE integration.
* Expanded the `README.md` with clear explanations of the auth-service's purpose, scope, functional requirements, API integration, and usage instructions, improving onboarding and developer understanding.

**Code Quality:**

* Introduced a new `.eslintrc.json` configuration enforcing code style (indentation, quotes, line length) and extending recommended ESLint and Google rules for improved code consistency and maintainability.

**Test Coverage:**

* Added unit tests for `loginEmployee`, `employeeChangePassword`, and `adminResetPassword` controllers, covering successful flows, error handling, and edge cases to ensure reliability and correctness of authentication logic. [[1]](diffhunk://#diff-3ba1099dbd5baa8a6d81d0773a3dd2bd649c683c9b431811e4131f665ab40d61R1-R152) [[2]](diffhunk://#diff-e78a8753a9b25dac4ea9e9dd4e6716d9cee74914ee99c5976202abac60f5d5c9R1-R174) [[3]](diffhunk://#diff-0df0ab04e6cb87870d0300747543257302f73d51369f384e0441f7450a8a5f48R1-R84)

## 📋 Checklist

* \[x] I have tested this code locally
* \[x] I have added/updated the unit tests to cover all my changes
* \[x] I have ensured my PR matched with related docs and contracts, and updated them if necessary
* \[x] Code is linted and formatted
* \[x] No console logs or commented-out code
* \[x] My code follows the [CONTRIBUTING.md](/CONTRIBUTING.md)

### CODE REVIEW

File: src/backend/user-service/src/controllers/getEmployeeDetail.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ * @param {import('express').Request} req - The Express request object.
+ * @param {object} req.params - The URL parameters.
+ * @param {string} req.params.employeeId - The ID of the employee whose details are to be fetched.
+ * @param {import('express').Response} res - The Express response object.
+ * @returns {Promise<void>} Sends a JSON response containing the employee's details or an error message.
+ */
+
+const getEmployeeDetail = async (req, res)=> {
+    try {
+        const {employeeId} = req.params;
```
[the remark this answers is not in the document]
Comment (perhanjay) [reply src/backend/user-service/src/controllers/getEmployeeDetail.js]: these alerts are irrelevant as it's a false alarm

File: src/backend/auth-service/src/app.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    methods: "GET,HEAD,PUT,PATCH,POST,DELETE", // Secara eksplisit izinkan metode
+};
+
+app.set("trust proxy", 1);
+// Satu masalah, kemungkinan API Gateway harus mencari cara mengubah header originnya agar bisa diteruskan
+app.use(cors(corsOptions));
+
+// CORS configuration for cross-origin requests
+app.use(express.json());
+app.use(cookieParser());
```
[the remark this answers is not in the document]
Comment (perhanjay) [reply src/backend/auth-service/src/app.js:81]: this comment is irrelevant as the csurf is already implemented in the right place

File: src/backend/user-service/src/services/urmService.js [hunk 3]
```diff
@@ -0,0 +1,377 @@
...
+        employees.position,
+        ARRAY_AGG(roles.name) AS roles
+      FROM employees
+      JOIN employee_roles ON employees.employeeId = employee_roles.employee_id
+      JOIN roles ON employee_roles.role_id = roles.roleId
+      WHERE employees.employeeId = $1
+      GROUP BY employees.full_name, employees.email, employees.department, employees.position;`;
+        const result = await pool.query(query, [employeeId]);
+
+        if (result.rows.length == 0) {
```
[the remark this answers is not in the document]
Comment (perhanjay) [reply src/backend/user-service/src/services/urmService.js]: Implemented

File: src/backend/user-service/src/controllers/editEmployeeDetails.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+const editEmployeeDetails = async (req, res) => {
+    const ALLOWED_USER_PATCH_FIELDS = ["full_name", "email", "department", "position", "is_active"];
+
+    const employeeId = req.params.employeeId;
+    const updates = req.body;
+
+    try {
+    // Cek apakah user ada (opsional, tapi disarankan untuk respons 404 yang akurat)
+        const checkUserResult = await findEmployeeById(employeeId);
+        if (checkUserResult.success == false) {
```
[the remark this answers is not in the document]
Comment (perhanjay) [reply src/backend/user-service/src/controllers/editEmployeeDetails.js]: implemented

File: src/backend/user-service/src/middlewares/authenticateServiceRequest.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+const authenticateServiceRequest = (options = {AdminRole: false, AuthService: false}) => {
+    return async (req, res, next) => {
+        const allowedRoles = [];
+
+        if (options.AdminRole == true) {
+            allowedRoles.push("SYSTEM_ADMIN");
+        }
+
+        if (options.AuthService == true) {
```
[the remark this answers is not in the document]
Comment (perhanjay) [reply src/backend/user-service/src/middlewares/authenticateServiceRequest.js]: implemented

File: src/backend/user-service/src/middlewares/authenticateServiceRequest.js [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+            return res.status(401).json({message: "Token format is 'Bearer <token>"});
+        }
+
+        try {
+            const decoded = jwt.verify(token, PUBLIC_KEY, {
+                algorithms: ["RS256"],
+                iss: "auth-service",
+            });
+
+            if (options.AdminRole == true || options.AuthService == true) {
```
[the remark this answers is not in the document]
Comment (perhanjay) [reply src/backend/user-service/src/middlewares/authenticateServiceRequest.js]: implemented

File: src/backend/user-service/src/routes/urm.routes.js [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+    authenticateServiceRequest({AuthService: true}),
+    createEmployee);
+
+
+/**
+ * @route POST /employee/get-employee
+ * @description Retrieves details of a specific employee by their ID.
+ * @access Protected (requires authentication)
+ */
+// CHANGES STARTS HERE DELETE LATER
```
[the remark this answers is not in the document]
Comment (perhanjay) [reply src/backend/user-service/src/routes/urm.routes.js]: implemented

File: src/backend/user-service/src/services/urmService.js [hunk 8]
```diff
@@ -0,0 +1,377 @@
...
+            return undefined;
+        }
+
+        return result.rows[0];
+    } catch (err) {
+        console.error(err);
```
[the remark this answers is not in the document]
Comment (perhanjay) [reply src/backend/user-service/src/services/urmService.js:130]: irrelevant

File: src/backend/user-service/src/services/urmService.js [hunk 9]
```diff
@@ -0,0 +1,377 @@
...
+        return {success: false, message: "Failed to insert employee details"};
+    } finally {
+        client.release(); // Release the client back to the pool
+    }
+};
+
+const findEmployeeById = async (employeeId) => {
+    const checkUserResult = await pool.query("SELECT * FROM employees WHERE employeeId = $1", [employeeId]);
+    if (checkUserResult.rows.length === 0) {
+        return {success: false, message: "pegawai tidak ditemukan."};
```
[the remark this answers is not in the document]
Comment (perhanjay) [reply src/backend/user-service/src/services/urmService.js:181]: irrelevant

File: src/backend/user-service/src/middlewares/authenticateServiceRequest.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (perhanjay) [reply src/backend/user-service/src/middlewares/authenticateServiceRequest.js]: implemented

File: src/backend/user-service/src/services/urmService.js [hunk 10]
```diff
@@ -0,0 +1,377 @@
...
+    `;
+        await client.query(employeeQuery, [employeeId, fullName, email, department, position]);
+
+        // Query the `roles` table to get the roleId
+        const roleQuery = `
+      SELECT roleId FROM roles WHERE name = $1
+    `;
+        const roleResult = await client.query(roleQuery, [roleName]);
+
+        if (roleResult.rows.length == 0) {
```
[the remark this answers is not in the document]
Comment (perhanjay) [reply src/backend/user-service/src/services/urmService.js:154]: implemented

**Review by andinaufal120 — APPROVED** [review]
# Initial merge of Auth Service and URM Service
Acnowledged, aprroved :heavy_check_mark:
